docs/source/troubleshooting.rst

```diff
@@ -18,10 +18,22 @@ Building
 Flashing
 --------
 
-* The board is not flashing.
+*Here are some of the tips if The board is not flashing..*
+
+* Sometimes to program ESP32 via serial you must keep GPIO0 LOW during the programming process.
+* Hold-down the **“BOOT”** button in your ESP32 board while unloading.
+* In some ESP32 development boards when trying to upload a new sketch, the Arduino IDE gives you a fatal error *(i.e A fatal error occurred: Failed to connect to ESP32: Timed out waiting for packet header).* This means your ESP32 board is not automatically in the flashing/uploading mode.
```

Comment:
This could be that your ESP32 board is not getting automatically in the upload/flash mode.

Reply:
When trying to upload a new sketch in some ESP32 development boards, the Arduino IDE gives you a fatal error *(i.e A fatal error occurred: Failed to connect to ESP32: Timed out waiting for packet header).* This means your ESP32 board is not automatically in the flashing/uploading mode.

Reply:
Please show me, What I can add/change or do better In this? @pedrominatel